### PR TITLE
Move core logic to hooks

### LINKED ISSUE #151: Getting "Warning: Cannot update a component from inside the function body of a different component." after upgrading to react v.16.13.1
## Information
- react-native version: 0.63.2
- react-navigation version: 5.4.2
- react-navigation-collapsible version: 5.6.1
- Platform (iOS/Android): iOS
- react-native init or Expo: react-native init 

## Detail
I get error "Warning: Cannot update a component from inside the function body of a different component." when calling  createCollapsibleStackSub() inside <FEStack.Navigator /> 

Screenshot 2020-07-29 at 17 07 08

my code:
`
function TabsScreen({ navigation }) {
  return (
    <FEStack.Navigator>
      {
       createCollapsibleStackSub(
        <FEStack.Screen
          name={sceneKeys.myScreen}
          component={MyScreen}
        />)
       }
    </FEStack.Navigator>
  );
}
`

Notice, that error disappear when I remov createCollapsibleStackSub() 

Might be related to https://github.com/facebook/react/issues/18178 and

### PR DESCRIPTION
**Pull request by benevbright**
Once suggested in #129 before
and resolves #151 #113

### CODE REVIEW

File: src/hooks.ts [hunk 1]
```diff
@@ -1,26 +1,129 @@
...
   const route = useRoute();
   const navigation = useNavigation();
 
-  const handleOrientationChange = () => {
-    navigation.setParams({ isCollapsibleDirty: true });
-  };
+  const positionY = React.useRef(new Animated.Value(0)).current;
+
   React.useEffect(() => {
+    const handleOrientationChange = () => {
```
Comment (joonhocho) [inline src/hooks.ts:48]: are these listeners necessary? cant you use [useWindowDimensions](https://reactnative.dev/docs/usewindowdimensions) and it watches size for you?

Comment (ozerty) [reply src/hooks.ts:48]: Indeed, it's now the preferred API to work with dimensions.

Comment (benevbright) [reply src/hooks.ts:48]: Hm, Interesting...
```
const { width, height } = useDimensions().window
const isLandscape = height < width
```
and use `isLandscape` at https://github.com/benevbright/react-navigation-collapsible/pull/155/files#diff-d5b97f4b9fc3ad944c938da8a20b475cR123? I'll try. Thanks.

Comment (joonhocho) [reply src/hooks.ts:48]:
```
const { width, height } = useWindowDimensions()
const isLandscape = height < width;
```

yeah. I think this should work and rerender when orientation changes.

Comment (benevbright) [reply src/hooks.ts]: https://github.com/benevbright/react-navigation-collapsible/pull/155/commits/164fa5d719b4bafa750c4609dfff0e99e7e3cda2 works perfectly. Thanks.

**Comment by benevbright** [comment]
Everything has been migrated to using hooks. I'll update README and then merge it.

**Pull request merged**